inc/Engine/Optimization/DelayJS/HTML.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace WP_Rocket\Engine\Optimization\DelayJS;
+
+use WP_Rocket\Admin\Options_Data;
+
+class HTML {
+
+	/**
+	 * Plugin options instance.
+	 *
+	 * @since  3.7
+	 *
+	 * @var Options_Data
+	 */
+	protected $options;
+
+	/**
+	 * Allowed scripts regex.
+	 *
+	 * @since 3.7
+	 *
+	 * @var string
+	 */
+	private $allowed_scripts = '';
+
+	/**
+	 * Creates an instance of HTML.
+	 *
+	 * @since  3.7
+	 *
+	 * @param Options_Data $options Plugin options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Adjust HTML to have delay js structure.
+	 *
+	 * @param string $html Buffer html for the page.
+	 *
+	 * @return string
+	 */
+	public function delay_js( $html ) {
+
+		if ( ! $this->is_allowed() ) {
+			return $html;
+		}
+
+		$this->allowed_scripts = $this->prepare_allowed_scripts_regex();
```

Comment:
We could bailout early here if there is no scripts in the list of allowed scripts, since nothing needs to happen. Small performance improvement